src/get_examples.py

```diff
@@ -18,10 +26,62 @@ def get_examples():
     ]
 
 
-def create_json():
+def get_examples_from_modified_files(modified_files: list[Path]):
+    """Get only examples for which files have been modified.
+
+    Parameters
+    ----------
+    modified_files :
+        List of modified files.
+
+        This can be obtained from something like ``git diff --name-only``.
+    """
+    modified_files = [file.resolve().relative_to(ROOT) for file in modified_files]
+
+    if any(
+        file.parts[0] != "examples" and str(file) not in IGNORE_CHANGES
+        for file in modified_files
+    ):
+        # This means that there is a modified file outside of the examples directory.
+        # In this case, all examples *could* be affected by the change.
+        return get_examples()
```

Comment:
@Luthaf This is the case in which we just can't tell which examples will get affected.

Reply:
thanks, I tried to find this but missed it!